src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/misc/ModuleAutoPearl.kt

```diff
@@ -0,0 +1,300 @@
+            owner = pearl.owner ?: player,
+            velocity = velocity,
+            pos = pearlPos
+        )?.pos ?: return
+
+        if (Limits.enabled && Limits.activationDistance > destination.distanceTo(player.pos)) {
+            return
+        }
+
+        val rotation = calculatePearlTrajectory(player.eyePos, destination) ?: return
+
+        if (!canThrow(rotation, destination)) {
+            return
+        }
+
+        if (queue.size() == 0) {
+            queue.add(rotation)
+        }
+    }
+
+    private fun canTrigger(pearl: EnderPearlEntity): Boolean {
+        if (Limits.enabled && Limits.angle < RotationManager.rotationDifference(pearl)) {
+            return false
+        }
+
+        if (pearl.owner == null) {
+            return mode == Modes.TRIGGER
+        }
+
+        if (pearl.ownerUuid == player.uuid) {
+            return false
+        }
+
+        return when(mode) {
+            Modes.TRIGGER -> pearl.owner!!.shouldBeAttacked()
+            Modes.TARGET -> ModuleKillAura.targetTracker.lockedOnTarget?.uuid == pearl.ownerUuid
+        }
+    }
+
+    private fun canThrow(
+        angles: Rotation,
+        destination: Vec3d
+    ): Boolean {
+        val simulatedDestination = TrajectoryInfoRenderer.getHypotheticalTrajectory(
+            entity = player,
+            trajectoryInfo = TrajectoryInfo.GENERIC,
+            rotation = angles
+        ).runSimulation(MAX_SIMULATED_TICKS)?.pos ?: return false
+
+        return !Limits.enabled || Limits.destDistance > destination.distanceTo(simulatedDestination)
+    }
+
+    private fun transformAngles(rotation: Rotation): Rotation {
+        var transformedYaw = rotation.yaw
+        var transformedPitch = rotation.pitch
+
+        transformedYaw = (transformedYaw % 360)
+        if (transformedYaw > 180) {
+            transformedYaw -= 360
+        } else if (transformedYaw < -180) {
+            transformedYaw += 360
+        }
+
+        transformedPitch = (transformedPitch % 360)
+        if (transformedPitch > 180) {
+            transformedPitch -= 360
+        } else if (transformedPitch < -180) {
+            transformedPitch += 360
+        }
+
+        if (transformedPitch > 90) {
+            transformedPitch = 180 - transformedPitch
+            transformedYaw += 180
+            transformedYaw = (transformedYaw % 360)
+            if (transformedYaw > 180) {
+                transformedYaw -= 360
+            } else if (transformedYaw < -180) {
+                transformedYaw += 360
+            }
+
+        } else if (transformedPitch < -90) {
+            transformedPitch = -180 - transformedPitch
+            transformedYaw += 180
+            transformedYaw = (transformedYaw % 360)
+            if (transformedYaw > 180) {
+                transformedYaw -= 360
+            } else if (transformedYaw < -180) {
+                transformedYaw += 360
+            }
+        }
+
+        return Rotation(transformedYaw, transformedPitch)
+    }
+
```

Comment:
What does this function do?

Reply:
My rewrite removes this function already - it is useless.

Reply:
> What does this function do?

yaw, pitch normalization 
-180 <= yaw <= 180, -90 <= pitch <= 90